### PR TITLE
P5-62 write about rts response time analysis and more

### PR DESCRIPTION
**Pull request by thomaslhansen**

### CODE REVIEW

File: report/3.Theory/2.RTS.tex [hunk 1]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
 
 This scheduling method can work both with and without pre-emption of tasks, meaning that a running task can be set on hold when a higher priority task is ready to run.
 When using this method for scheduling, a task with a low priority is allowed to run if no other task with a higher priority is ready to run. 
-Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
\ No newline at end of file
+Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
+
+\section{Response time analysis}
+Response time analysis is an exact schedulability test for an fixed-priority  assignment scheme on a single-processor system.
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: for a~~n~~ fixed-priority

File: report/3.Theory/2.RTS.tex [hunk 2]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
 This scheduling method can work both with and without pre-emption of tasks, meaning that a running task can be set on hold when a higher priority task is ready to run.
 When using this method for scheduling, a task with a low priority is allowed to run if no other task with a higher priority is ready to run. 
-Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
\ No newline at end of file
+Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
+
+\section{Response time analysis}
+Response time analysis is an exact schedulability test for an fixed-priority  assignment scheme on a single-processor system. 
+The response time analysis allows for prediction of the worst-case response time for each task which depends on the interference due to execution of higher priority tasks. 
+The worst-case response times are then compared to with the corresponding tasks deadlines to evaluate whether all tasks meets their deadlines.
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: ... compared **to with** the ...

Comment (technight) [reply report/3.Theory/2.RTS.tex]: meet~~s~~ - I think

File: report/3.Theory/2.RTS.tex [hunk 3]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+\begin{equation}
+\ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Since all tasks with a higher priority has an impact on the worst-case interference of a lower priority task all the higher priority tasks interference needs to be taken into account thus resulting in the following equation:
+\begin{equation}
+I_i = \sum_{j\in hp(i)} \cdot \ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Then recalling \ref{eq:responsetime} it is seen we get the following recursive relation for the worst-case response time $R_i$ for $\tau_i$:
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: "Then recalling ... it is seen we get" - Sounds funny, could it be rephrased?

File: report/3.Theory/2.RTS.tex [hunk 4]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+\begin{equation}
+R_i = C_i + \sum_{j\in hp(i)} \cdot \ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Since this is a recursive definition the equation may have more then one solution, hence the smallest solution to the equation is also the actual worst-case response time. 
+The simplest way of solving this recursive equation is be transforming it into the form of a recurrence relationship as follows where $hp(i)$ is any higher priority task:
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: equation is **by** transforming

File: report/3.Theory/2.RTS.tex [hunk 5]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+\begin{equation}
+w^{(k+1)}_{i} = C_i + \sum_{j\in hp(i)} \cdot \ceil*{\frac{w^{k}_{i}}{T_j}} \cdot C_j
+\end{equation}
+When using this recurrence relationship to evaluate the worst-case response time the result from the latest evaluation is feed back into the recurrence relationship until a fix point has been found where the input is equal to the output.
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: feed -> fed i think

File: report/3.Theory/2.RTS.tex [hunk 6]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+\end{equation}
+
+Since this is a recursive definition the equation may have more then one solution, hence the smallest solution to the equation is also the actual worst-case response time. 
+The simplest way of solving this recursive equation is be transforming it into the form of a recurrence relationship as follows where $hp(i)$ is any higher priority task:
+\begin{equation}
+w^{(k+1)}_{i} = C_i + \sum_{j\in hp(i)} \cdot \ceil*{\frac{w^{k}_{i}}{T_j}} \cdot C_j
+\end{equation}
+When using this recurrence relationship to evaluate the worst-case response time the result from the latest evaluation is feed back into the recurrence relationship until a fix point has been found where the input is equal to the output. 
+
+When the worst-case response time has been calculated for all given tasks of a system, the worst-case response times is verified against the deadlines of the task it self.
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: times are verified / time is verified

File: report/3.Theory/2.RTS.tex [hunk 7]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+If any worst-case response time of a task is greater then the deadline of that task the task is not schedulable, hence the entire system is not schedulable.
```
Comment (technight) [inline report/3.Theory/2.RTS.tex]: greater than

File: report/3.Theory/2.RTS.tex [hunk 8]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
 
 This scheduling method can work both with and without pre-emption of tasks, meaning that a running task can be set on hold when a higher priority task is ready to run.
 When using this method for scheduling, a task with a low priority is allowed to run if no other task with a higher priority is ready to run. 
-Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
\ No newline at end of file
+Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
```
Comment (nillas12) [inline report/3.Theory/2.RTS.tex]: Isn't it: "Given that" Syntes i hvertfald at der er et eller andet galt i begyndelsen af sætningen

File: report/3.Theory/2.RTS.tex [hunk 9]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+
+\section{Response time analysis}
+Response time analysis is an exact schedulability test for an fixed-priority  assignment scheme on a single-processor system. 
+The response time analysis allows for prediction of the worst-case response time for each task which depends on the interference due to execution of higher priority tasks. 
+The worst-case response times are then compared to with the corresponding tasks deadlines to evaluate whether all tasks meets their deadlines. 
+
+Doing execution of a task, the task can be pre-empted if a higher priority task is released.
```
Comment (nillas12) [inline report/3.Theory/2.RTS.tex]: Is this supposed to be during? Or am I just being bad.

File: report/3.Theory/2.RTS.tex (hunk 1 above)
Comment (Moesgaarda) [reply report/3.Theory/2.RTS.tex]: DOBBELT MELLEMRUM!!!!!!!!!

File: report/3.Theory/2.RTS.tex (hunk 3 above)
Comment (Moesgaarda) [reply report/3.Theory/2.RTS.tex]: Skal der måske stå when?

File: report/3.Theory/2.RTS.tex [hunk 10]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+\end{equation}
+
+Since this is a recursive definition the equation may have more then one solution, hence the smallest solution to the equation is also the actual worst-case response time. 
+The simplest way of solving this recursive equation is by transforming it into the form of a recurrence relationship as follows where $hp(i)$ is any higher priority task:
+\begin{equation}
+w^{(k+1)}_{i} = C_i + \sum_{j\in hp(i)} \cdot \ceil*{\frac{w^{k}_{i}}{T_j}} \cdot C_j
+\end{equation}
+When using this recurrence relationship to evaluate the worst-case response time the result from the latest evaluation is fed back into the recurrence relationship until a fix point has been found where the input is equal to the output. 
+
+When the worst-case response time has been calculated for all given tasks of a system, the worst-case response times are verified against the deadlines of the task it self.
```
Comment (Moesgaarda) [inline report/3.Theory/2.RTS.tex]: itself

File: report/3.Theory/2.RTS.tex [hunk 11]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+\begin{equation}
+\ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Since all tasks with a higher priority has an impact on the worst-case interference of a lower priority task all the higher priority tasks interference needs to be taken into account thus resulting in the following equation:
+\begin{equation}
+I_i = \sum_{j\in hp(i)} \cdot \ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Then combined with Equation~\ref{eq:responsetime} it is seen we get the following recursive relation for the worst-case response time $R_i$ for $\tau_i$:
```
Comment (Moesgaarda) [inline report/3.Theory/2.RTS.tex]: When?

File: report/3.Theory/2.RTS.tex [hunk 12]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
-Given a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
\ No newline at end of file
+Given that a low priority task is running and a higher priority task is marked as ready, the low priority task will be pre-empted in order to allow the higher priority to run.
+
+\section{Response time analysis}
+Response time analysis is an exact schedulability test for a fixed-priority assignment scheme on a single-processor system. 
+The response time analysis allows for prediction of the worst-case response time for each task which depends on the interference due to execution of higher priority tasks. 
+The worst-case response times are then compared to the corresponding tasks deadlines to evaluate whether all tasks meets their deadlines. 
+
+Doing execution of a task, the task can be pre-empted if a higher priority task is released thereby delaying the response time of the task.
```
Comment (Moesgaarda) [inline report/3.Theory/2.RTS.tex]: During?

File: report/3.Theory/2.RTS.tex [hunk 13]
```diff
@@ -44,4 +44,42 @@ \subsubsection{Task-based scheduling}
...
+R_i = C_i + I_i
+\end{equation}
+
+The contribution of each higher priority task to the overall worst-case interference has to be evaluated individually by considering the interference of a single task $\tau_j$ of higher priority than $\tau_i$.
+The exact number of releases can be computed by means of the ceiling function and since each release of $\tau_J$ will impose on $\tau_i$ an interference of $C_j$ the worst-case interference imposed on $tau_i$ by $\tau_j$ is as follows:
+\begin{equation}
+\ceil*{\frac{R_i}{T_j}} \cdot C_j
+\end{equation}
+
+Since all tasks with a higher priority has an impact on the worst-case interference of a lower priority task all the higher priority tasks interference needs to be taken into account thus resulting in the following equation:
```
Comment (Moesgaarda) [inline report/3.Theory/2.RTS.tex]: has skal være have grundet flertal.

needs skal være uden s, da det er flertal

**Pull request merged**